tidy3d/components/base.py

```diff
@@ -128,12 +128,18 @@ def __hash__(self) -> int:
         except TypeError:
             return hash(self.json())
```

Comment:
Note that at least since the introduction of `Tidy3dBaseModel.attrs`, `__hash__` will _always_ hash the json string (because dicts are not hashable), so the `try/except` here is somewhat superfluous. Should we do something about that @tylerflex?